src/main/java/org/openstreetmap/atlas/geography/sharding/Sharding.java

```diff
@@ -81,39 +76,11 @@ static Sharding forString(final String sharding)
     /**
      * Generate shards. This needs to be deterministic!
      *
-     * @param multiPolygon
+     * @param surface
      *            The bounds to limit the shards.
      * @return The shards {@link Iterable}.
      */
-    default Iterable<? extends Shard> shards(final MultiPolygon multiPolygon)
```

Comment:
Why is there no longer a default?

Reply:
Both Polygon and MultiPolygon are GeometricSurfaces. That interface contains the contract for the overlapping logic that is in this default function. Ideally, all MultiPolygon intersection is handled by the shape class and no implementation knowledge is needed. Deleting this removes duplicate work and prevents it breaking if MultiPolygon implementation is ever changed.